cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-scaling.adoc

```diff
@@ -0,0 +1,269 @@
+= Tutorial: Scaling an application
+include::_attributes/attributes-openshift-dedicated.adoc[]
+:context: cloud-experts-deploying-application-scaling
+:source-highlighter: pygments
+:pygments-style: emacs
+:icons: font
+
+toc::[]
+
+//rosaworkshop.io content metadata
+//Brought into ROSA product docs 2024-04-10
+
+== Scaling
+
+You can manually scale your pods or you can automatically scale your pods by using the Horizontal Pod Autoscaler (HPA). You can also scale your cluster nodes.
+
+=== Manual pod scaling
+
+You can manually scale the pods of your applications:
+
+* by changing your replicaset or deployment definition 
+* by using the command line
+* by using the web console
+
+This workshop starts by using only one pod for the microservice. By defining a replica of `1` in our deployment definition, the Kubernetes Replication Controller strives to keep one pod alive. You can also define pod autoscaling by using the link:https://docs.openshift.com/container-platform/latest/nodes/pods/nodes-pods-autoscaling.html[Horizontal Pod Autoscaler](HPA) which is based on the load and will scale out more pods, beyond our initial definition, if high load is experienced.
+
+.Prerequisites
+
+* Created a ROSA cluster 
+* Have successfully deloyed the OSToy application
+
+.Procedure 
+
+. In the OSToy app, click the *Networking* tab in the left menu.
+. In the "Intra-cluster Communication" section, there is a box that randomly changes colors. Inside the box, you see the microservice's pod name. There is only one box shown because there is only one microservice pod. 
++
+image::deploy-scale-network.png[HPA Menu]
++ 
+
+. Run the following command to confirm that there is only one pod running for the microservice:
++
+[source,terminal,highlight='4']
+----
+$ oc get pods
+----
++
+.Example output
+[source,terminal]
+----
+NAME                                  READY     STATUS    RESTARTS   AGE
+ostoy-frontend-679cb85695-5cn7x       1/1       Running   0          1h
+ostoy-microservice-86b4c6f559-p594d   1/1       Running   0          1h
+----
+
+. Download the link:https://www.rosaworkshop.io/ostoy/yaml/ostoy-microservice-deployment.yaml[ostoy-microservice-deployment.yaml] and save it to your local machine.
+. Change the deployment definition to three pods instead of one, similar to the following example:
++
+[source,yaml]
+----
+spec:
+    selector:
+      matchLabels:
+        app: ostoy-microservice
+    replicas: 3
+----
+
+. Now, scale the application by using the CLI or using the Web UI:
++
+** Run the following command to increase the number of pods from `1` to `3`:
++
+[source,terminal]
+----
+$ oc scale deployment ostoy-microservice --replicas=3
+----
++
+** In the OpenShift web console, from the left menu, click *Workloads > Deployments > ostoy-microservice*.  On the left side of the page, you see a blue circle with a "1 Pod" label in the middle. Selecting the arrows next to this scales the number of pods. Select the up arrow to `3`.
```

Comment:
If we followed this flow we would already be at 3 pods and clicking twice on that arrow would give us 5 pods.  We need to look at the flow as a whole.